prov/util/src/util_mr_cache.c

```diff
@@ -89,13 +96,7 @@ static void util_mr_free_entry(struct ofi_mr_cache *cache,
 					entry->info.iov.iov_len);
```

Comment:
Since this unsubscribe is still hit by the userfault notify path, it won't fix #5347

I wrote a tiny workaround patch for testing and created a util_mr_uncache_entry_notify function that does the same thing but without an unsubscribe call and it fixed the merge region bug for me.

Maybe we can have a boolean parameter in util_mr_uncache_entry whether to unsubscribe a region or not. This can be set by whatever monitor is being used (So userfault would pass 0 and other monitors may pass 1).

Reply:
Moving the unsubscribe from the free path to the uncache path changes the behavior.  So I wouldn't assume that the existence of calling unsubscribe is the cause of the error.  We _should_ be able to call unsubscribe and have correct behavior.  Can you test with this patch applied?

Reply:
I've already tested with this patch applied. It fails with merge regions still.

If I change the ofi_mr_cache_notify to call a different function (util_mr_cache_entry_notify - does the same thing w/o the unsubscribe), it works.

Reply:
Okay - I still don't understand why we have the failure.  There must still be some race.  If we can't find it, I agree that we remove the uffd unregister call